nodebuilder/settings.go

```diff
@@ -21,10 +35,71 @@ func WithBootstrappers(peers params.Bootstrappers) fx.Option {
 }
 
 // WithMetrics enables metrics exporting for the node.
-func WithMetrics(enable bool) fx.Option {
-	return fx.Options(
-		// TODO: Add metrics to more modules and enable here.
-		headermodule.WithMetrics(enable),
-		daser.WithMetrics(enable),
+func WithMetrics(enable bool, metricOpts []otlpmetrichttp.Option, nodeType node.Type) fx.Option {
+	if !enable {
+		return fx.Options()
+	}
+
+	baseComponents := fx.Options(
+		fx.Supply(metricOpts),
+		fx.Invoke(InitializeMetrics),
+		fx.Invoke(header.MonitorHead),
+		fx.Invoke(state.MonitorPFDs),
+	)
+
+	var opts fx.Option
+	switch nodeType {
+	case node.Full:
```

Comment:
is there a reason to separate case node.Full and case node.Light? Unless it is expected to add diff metrics for full and light in the near future.

Reply:
done